analytic_engine/src/sst/factory.rs

```diff
@@ -104,11 +99,13 @@ impl Factory for FactoryImpl {
         &self,
         options: &SstReaderOptions,
         path: &'a Path,
+        storage_format: StorageFormat,
         store_picker: &'a ObjectStorePickerRef,
     ) -> Option<Box<dyn SstReader + Send + 'a>> {
         // TODO: Currently, we only have one sst format, and we have to choose right
         // reader for sst according to its real format in the future.
-        let reader = AsyncParquetReader::new(path, store_picker, options);
+        let hybrid_encoding = matches!(storage_format, StorageFormat::Hybrid);
+        let reader = AsyncParquetReader::new(path, hybrid_encoding, store_picker, options);
```

Comment:
why not pass StorageFormat into AsyncParquetReader?

What if we add a third format besides columnar/hybrid, but they are all based on parquet

Reply:
`AsyncParquetReader` should not have knowledge about the `StorageForamt`, especially, in the future, some other format will be introduced.

> What if we add a third format besides columnar/hybrid, but they are all based on parquet
As for the third format based on parquet, I guess we should define a different reader rather than reusing the `AsyncParquetReader`.

Reply:
Make sense.